### PR TITLE
Refactoring vuex cms module

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related issues
closes #3337 

### Short description and why it's useful
Refactoring vuex cms module

### Which environment this relates to
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance
- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

**Comment by pkarw** [comment]
Hints for @alinadivante and QA: we should test the static pages links in the footer if they still works correctly

File: core/modules/cms/store/block/getters.ts [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+
+    return getters.cmsBlockIdentifier(identifier)
   },
-  cmsBlockId: (state) => (id) => {
-    return state.items.find(item => item.id === id)
+  cmsBlockId: (state, getters) => (id) => {
+    Logger.error('The getter cmsBlock/cmsBlockId has been deprecated please change to cmsBlock/getCmsBlockId')()
+
+    return getters.getCmsBlockId(id)
+  },
```
Comment (pkarw) [inline core/modules/cms/store/block/getters.ts]: Shouldn’t it be getCmsBlocById / I mean the adding “by” we will end up with improved grammar; well it’s just a proposition ;)

File: core/modules/cms/store/block/getters.ts [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
 import CmsBlockState from '../../types/CmsBlockState'
 import RootState from '@vue-storefront/core/types/RootState'
+import { Logger } from '@vue-storefront/core/lib/logger'
 
 const getters: GetterTree<CmsBlockState, RootState> = {
-  cmsBlocks: (state) => state.items,
-  cmsBlockIdentifier: (state) => (identifier) => {
-    return state.items.find(item => typeof item === 'object' && item.identifier === identifier)
+  cmsBlocks: (state, getters) => {
+    Logger.error('The getter cmsBlock/addMethod has been deprecated please change to cmsBlock/getCmsBlocks')()
```
Comment (patzick) [inline core/modules/cms/store/block/getters.ts]: unfortunately this is invoked not on usage but always in the application, even if not used. We should remove this getters and add instruction to upgrade notes to replace them to new ones around application.

File: core/modules/cms/store/block/getters.ts [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
+    return getters.getCmsBlockById(id)
+  },
+  getCmsBlockByIdentifier: (state) => (identifier) =>
+    state.items.find(item => typeof item === 'object' && item.identifier === identifier),
+  getCmsBlockById: (state) => (id) => state.items.find(item => item.id === id),
+  getCmsBlocks: (state) => state.items,
+  hasItems: (state) => state.items && state.items.length > 0,
+  findCmsBlocks: (state, getters) => ({ key, value }) => {
+    if (getters.hasItems) {
+      return state.items.filter(item => item[key] === value)
```
Comment (patzick) [inline core/modules/cms/store/block/getters.ts]: you could do only this and just ensure in mutations that state.items is always an array

File: core/modules/cms/store/page/getters.ts [hunk 4]
```diff
@@ -1,9 +1,17 @@
 import { GetterTree } from 'vuex'
 import RootState from '@vue-storefront/core/types/RootState'
 import CmsPageState from '../../types/CmsPageState'
+import { Logger } from '@vue-storefront/core/lib/logger'
 
 const getters: GetterTree<CmsPageState, RootState> = {
-  cmsPages: (state) => state.items
+  cmsPages: (state, getters) => {
+    Logger.error('The getter cmsPage/cmsPages has been deprecated please change to cmsPage/getCmsPages')()
```
Comment (patzick) [inline core/modules/cms/store/page/getters.ts]: same here with loggers

**Pull request merged**